### PR TITLE
Puppeteer: minor test code refactor

### PR DESCRIPTION
**Pull request by aweng98**

### CODE REVIEW

File: e2e/tests/workspace/cdr-version-display.spec.ts [hunk 1]
```diff
@@ -10,17 +10,6 @@ describe('Workspace CDR Version display', () => {
         await signIn(page);
     });
 
-    test('The CDR version displays in the workspace navigation bar', async () => {
```
Comment (aweng98) [inline e2e/tests/workspace/cdr-version-display.spec.ts:13]: checking cdr version on Data page is moved to `workspace-create.spec` test (one less new workspace).

File: e2e/tests/workspace/workspace-edit.spec.ts [hunk 2]
```diff
@@ -85,8 +85,7 @@ describe('Editing workspace via workspace card snowman menu', () => {
 
   test('User as OWNER can edit workspace via workspace action menu', async () => {
     const workspaceCard = await findOrCreateWorkspace(page);
-    const workspaceName = await workspaceCard.getWorkspaceName();
-    console.log(workspaceName);
```
Comment (aweng98) [inline e2e/tests/workspace/workspace-edit.spec.ts:89]: not useful, cluttering code.

File: e2e/jest-circus.setup.ts [hunk 3]
```diff
@@ -2,4 +2,4 @@ require('expect-puppeteer');
 
 // Runs failed tests n-times until they pass or until the max number of retries is exhausted. This only works with jest-circus.
 // This file loaded in jest.config.js ==> setupFilesAfterEnv
-jest.retryTimes(process.env.DEBUG === 'true' ? 0 : 1);
+// jest.retryTimes(process.env.DEBUG === 'true' ? 0 : 1);
```
Comment (aweng98) [inline e2e/jest-circus.setup.ts:5]: disable global retry failed tests in an indiscriminate fashion. Retry uses up billing projects and may hides application issues.

File: e2e/package.json [hunk 4]
```diff
@@ -9,10 +9,10 @@
     "test": "cross-env jest --maxWorkers=5",
     "test:debugTest": "cross-env PUPPETEER_HEADLESS=false DEBUG=true node --inspect-brk node_modules/.bin/jest",
     "test:debug": "cross-env PUPPETEER_HEADLESS=false DEBUG=true jest --detectOpenHandles",
-    "test-local": "cross-env WORKBENCH_ENV=local DEBUG=true jest --maxWorkers=3",
-    "test-local:debug": "cross-env WORKBENCH_ENV=local PUPPETEER_HEADLESS=false DEBUG=true jest --maxWorkers=3",
+    "test-local": "cross-env WORKBENCH_ENV=local DEBUG=true jest --detectOpenHandles",
```
Comment (aweng98) [inline e2e/package.json:12]: `--maxWorkers=3` will cause test to fail when only one test is selected to run.

Comment (jmthibault79) [reply e2e/package.json:12]: This is surprising.  Where have you seen this?

Also: why change some of these but not others, and what does adding `detectOpenHandles` do?

**Review by jmthibault79 — COMMENTED** [review]
Code changes look good.  I'm not sure about the package.json changes.

File: e2e/package.json (hunk 4 above)
Comment (aweng98) [reply e2e/package.json:12]: The error is `Protocol error (DOM.describeNode): Target closed.` when running only one test on localhost like `yarn test-local tests/home/home-page.spec.ts`.
`--detectOpenHandles` detects one code issue: when `asyn/promise` code didn't use `await`.

**Review by aweng98 — COMMENTED** [review]
Thanks for reviewing @jmthibault79

File: e2e/tests/datasets/dataset-create.spec.ts [hunk 5]
```diff
@@ -21,8 +21,7 @@ describe('Dataset test', () => {
    * - Edit dataset. Save dataset without Export to Notebook.
    * - Delete Dataset.
    */
-  // RW-5751 Search for condition is failing in CI.
-  xtest('Can create Dataset with user-defined Cohorts', async () => {
+  test('Can create Dataset with user-defined Cohorts', async () => {
```
Comment (aweng98) [inline e2e/tests/datasets/dataset-create.spec.ts:24]: [RW-5751](https://precisionmedicineinitiative.atlassian.net/browse/RW-5751) is done but no one re-enabled test.

File: e2e/package.json (hunk 4 above)
Comment (jmthibault79) [reply e2e/package.json:12]: I tried this out, and I was able to run a single test with maxWorkers=3.  Let's keep that flag.
```
thibault@wm98d-e39 ~/src/aou/e2e (joel/create-auth-domain) $ yarn test-local tests/home/home-page.spec.ts                                                                                                                                           yarn run v1.21.1                                                                                                                                                                                                                                    $ cross-env WORKBENCH_ENV=local DEBUG=true jest --maxWorkers=3 tests/home/home-page.spec.ts                                                                                                                                                          PASS  tests/home/home-page.spec.ts (35.393 s)                                                                                                                                                                                                        Home page ui tests                                                                                                                                                                                                                                    ✓ Check visibility of Workspace cards (16831 ms)                                                                                                                                                                                                    ✓ Check Create New Workspace link on Home page (15654 ms)                                                                                                                                                                                                                                                                                                                                                                                                                                           Test Suites: 1 passed, 1 total                                                                                                                                                                                                                      Tests:       2 passed, 2 total                                                                                                                                                                                                                      Snapshots:   0 total                                                                                                                                                                                                                                Time:        36.542 s                                                                                                                                                                                                                               Ran all test suites matching /tests\/home\/home-page.spec.ts/i.                                                                                                                                                                                     ✨  Done in 39.23s.
```

Comment (jmthibault79) [reply e2e/package.json:12]: `detectOpenHandles` looks useful.  Thanks 👍

Comment (aweng98) [reply e2e/package.json]: Reverted package.json back to original, saving `detectOpenHandles` change for another day.

**Review by jmthibault79 — APPROVED** [review]

**Pull request merged**